cmd/intake-receiver/main.go

```diff
@@ -0,0 +1,331 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"compress/zlib"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"go.elastic.co/apm/v2/model"
+)
+
+var maxScannerBufSize = 300 * 1024 // APM Server default
+
+func main() {
+	// Ignored flags, they are just here to allow the `intake-receiver` to be
+	// dropped in as a replacement for APM Server. This means, that all the
+	// config options are ignored.
+	flag.String("e", "", "apm-server compatibility option")
+	flag.String("E", "", "apm-server compatibility option")
+	flag.String("httpprof", "", "apm-server compatibility option")
+
+	var host, version, folder string
+	flag.StringVar(&host, "host", ":8200", "port that the server will listen to")
+	// TODO(marclop) how to keep this version in sync with `../cmd/version.go`?
+	flag.StringVar(&version, "version", "8.2.0", "APM Server version")
```

Comment:
Why does this need to be configurable; shouldn't it always be aligned with the server version of the branch? 

As your comment points out, this should be automatically kept in sync with the apm-server version. How about updating the version with the `make update` command for now? (for example, writing to a `version.go` file and then reading that from here)